test/platform-stats.test.js

```diff
@@ -226,6 +235,191 @@ describe('platform-stats', () => {
     })
   })
 
+  describe('refreshDatabase', () => {
+    it('runs provided functions and handles errors', async () => {
+      const executedFunctions = []
+      const errorMessages = []
+
+      const successFunction = async () => {
+        executedFunctions.push('successFunction')
+      }
+
+      const errorFunction = async () => {
+        executedFunctions.push('errorFunction')
+        throw new Error('Test error')
+      }
+
+      const originalConsoleError = console.error
+      console.error = (message, error) => {
+        errorMessages.push({ message, error })
+      }
+
+      await refreshDatabase(createPgClient, {
+        functionsToRun: [successFunction, errorFunction]
+      })
+
+      console.error = originalConsoleError
+
+      assert.deepStrictEqual(executedFunctions, ['successFunction', 'errorFunction'])
+      assert.strictEqual(errorMessages.length, 1)
+      assert.strictEqual(errorMessages[0].message, 'Error running function errorFunction:')
+      assert.strictEqual(errorMessages[0].error.message, 'Test error')
+    })
+  })
+
+  describe('updateStationsAndParticipants', () => {
+    it('updates recent_station_details, recent_active_stations, and recent_participant_subnets', async () => {
+      const participantsMap = await mapParticipantsToIds(pgClient, new Set(['0x10', '0x20']))
+
+      /** @type {Measurement[]} */
+      const honestMeasurements = [
+        { ...VALID_MEASUREMENT, stationId: 'station1', participantAddress: '0x10', inet_group: 'subnet1' },
+        { ...VALID_MEASUREMENT, stationId: 'station1', participantAddress: '0x10', inet_group: 'subnet2' },
+        { ...VALID_MEASUREMENT, stationId: 'station2', participantAddress: '0x20', inet_group: 'subnet3' }
+      ]
+
+      /** @type {Measurement[]} */
+      const allMeasurements = [
+        ...honestMeasurements,
+        { ...VALID_MEASUREMENT, stationId: 'station1', participantAddress: '0x10', inet_group: 'subnet1', fraudAssessment: 'TASK_NOT_IN_ROUND' }
+      ]
+
+      await updateStationsAndParticipants(pgClient, honestMeasurements, allMeasurements, participantsMap, today)
+
+      const { rows: stationDetails } = await pgClient.query(`
+        SELECT
+          day::TEXT,
+          station_id,
+          participant_id,
+          accepted_measurement_count,
+          total_measurement_count
+        FROM recent_station_details
+        WHERE day = $1::DATE
+        ORDER BY station_id, participant_id
+      `, [today])
+
+      assert.strictEqual(stationDetails.length, 2)
+      assert.deepStrictEqual(stationDetails, [
```

Comment:
`deepStrictEqual` will implicitly assert the array size and it will provide a more descriptive error message. I propose removing the first assertion.

```suggestion
      assert.deepStrictEqual(stationDetails, [
```

Reply:
(The same comment applies to other places asserting array length first and array contents second.)